stan/math/fwd.hpp

```diff
@@ -5,9 +5,9 @@
 
 #include <stan/math/fwd/core.hpp>
 #include <stan/math/fwd/meta.hpp>
-#include <stan/math/prim.hpp>
-
 #include <stan/math/fwd/fun.hpp>
 #include <stan/math/fwd/functor.hpp>
 
+#include <stan/math/prim.hpp>
```

Comment:
This seems pretty innocuous but why'd it move?

Reply:
If a prim function is used for fwd autodiff and that prim function calls another function that has fwd implementation, we need to include that fwd implementation before the prim function. Since no prim functions include fwd stuff, it is the easiest to include all fwd stuff before any prim stuff.

Reply:
This makes sense to me. I just moved a header today cuz of this I think.

Reply:
That makes sense...

for a while, we were trying to get it where once you included the top-level includes for any of the prim / rev / fwd / mix, then everything else is not order-dependent. So... the rest was left alphabetical.

But it was really hard to keep that way.